sdk/java-sdk/src/main/scala/kalix/javasdk/impl/MetadataImpl.scala

```diff
@@ -250,6 +249,19 @@ object MetadataImpl {
 
   val Empty = new MetadataImpl(Vector.empty)
 
+  def of(metadata: Metadata): MetadataImpl = {
+    metadata
+      .iterator()
+      .asScala
+      .foldLeft(MetadataImpl.Empty)((meta, entry) => {
+        if (entry.isText) {
+          meta.add(entry.getKey, entry.getValue)
+        } else {
+          meta.addBinary(entry.getKey, entry.getBinaryValue)
+        }
+      })
+  }
```

Comment:
It is needed only to avoid duplication of code in `GrpcDeferredCall` and `HttpDeferredCall`, unless I will be brutal and do sth like:
```
override def withMetadata(metadata: Metadata): GrpcDeferredCall[I, O] = {
    this.copy(metadata = metadata.asInstanceOf[MetadataImpl])
  }
```

Reply:
I think brutal would be ok here

Reply:
(since the interface is not really an extension point but rather a way to hide impl details)

Reply:
done